ciao-controller/instance.go

```diff
@@ -153,12 +164,32 @@ func (c *config) GetResources() map[string]int {
 	return resources
 }
 
-func getStorage(c *controller, wl *types.Workload, tenant string, instanceID string) (payloads.StorageResources, error) {
-	s := wl.Storage
+func addBlockDevice(c *controller, tenant string, instanceID string, device storage.BlockDevice, s types.StorageResource) (payloads.StorageResource, error) {
+	// don't you need to add support for indicating whether
+	// a block device is bootable.
+	data := types.BlockData{
+		BlockDevice: device,
+		Size:        s.Size,
+		CreateTime:  time.Now(),
+		TenantID:    tenant,
+		Name:        fmt.Sprintf("Storage for instance: %s", instanceID),
+		Description: s.Tag,
+	}
+
+	err := c.ds.AddBlockDevice(data)
+	if err != nil {
+		c.DeleteBlockDevice(device.ID)
+		return payloads.StorageResource{}, err
+	}
+
+	return payloads.StorageResource{ID: data.ID, Bootable: s.Bootable, Ephemeral: s.Persistent}, nil
```

Comment:
`Ephemeral: s.Persistent`

looks wrong. Is there a logical not missing somewhere?